runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6867,8 +6880,16 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
       //
       TR::IlGeneratorMethodDetails & details = entry->getMethodDetails();
 
-      eligibleForRelocatableCompile =
-            (enableJITaaSRemoteAOT || !entry->isOutOfProcessCompReq()) && // disable remote JITaaS AOT until it's working
+      if (_compInfo.getPersistentInfo()->getJITaaSMode() == SERVER_MODE)
+         {
+         // Since all of the preliminary checks have already been done at the client,
+         // eligibleForRelocatableCompile should be true at the server side if the client
+         // requests AOT.
+         eligibleForRelocatableCompile = (entry->_useAotCompilation) ? true : false;
```

Comment:
`eligibleForRelocatableCompile = entry->_useAotCompilation`  since `_useAotCompilation` is already a bool